File: ruby_scripts/find_url.rb
require 'httparty'
require 'nokogiri'

url = 'https://tns4lpgmziiypnxxzel5ss5nyu0nftol.lambda-url.us-east-1.on.aws/challenge'
response = HTTParty.get(url)

if response.code == 200
    html_text = Nokogiri::HTML(response.body)

    new_url = ""

    html_text.css('code').each do |code_tag|
    if code_tag['data-class']&.start_with?('23')
      # Find the <div> tag within the <code> tag with 'data-tag' ending in "93"
      div_tag = code_tag.at_css('div[data-tag$="93"]')
      if div_tag
        # Find the <span> tag within the <div> tag with 'data-id' containing "21"
        span_tag = div_tag.at_css('span[data-id*="21"]')
        if span_tag
          # Find the <i> tag within the <span> tag
          i_tag = span_tag.at_css('i')
          if i_tag
            # Append the text of the <i> tag to the URL segments
            url_segments += i_tag.text
          end
        end
      end
    end
  end

  puts "New URL: #{new_url}"
else
    puts 'Connection failed'
end

File: python_scripts/find_url.py
"""Script to decode hidden link"""

import requests
from bs4 import BeautifulSoup

# Fetch the webpage content
url = 'https://tns4lpgmziiypnxxzel5ss5nyu0nftol.lambda-url.us-east-1.on.aws/challenge'
response = requests.get(url)

# Check to make sure the request was successful
if response.status_code == 200:
    # Parse the HTML content
    soup = BeautifulSoup(response.text, 'html.parser')

    # Initialize an empty string to store the URL segments
    url_segments = ""

    for code_tag in soup.find_all('code', {'data-class': lambda c: c and c.startswith('23')}):
        div_tag = code_tag.find('div', {'data-tag': lambda x: x and x.endswith('93')})
        if div_tag:
            span_tag = div_tag.find('span', {'data-id': lambda x: x and "21" in x})
            if span_tag:
                i_tag = span_tag.find('i')
                if i_tag:
                    # Append value attribute of the <i> tag to the URL segments
                    url_segments += i_tag.get('value')

    print(f"New URL: {url_segments}")
else:
    print(f"Failed to retrieve the webpage. Status code: {response.status_code}")
